src/main/java/StringManipulation.java

```diff
@@ -1,33 +1,97 @@
 public class StringManipulation implements StringManipulationInterface {
 
+    private String string;
     @Override
     public String getString() {
-        return null;
+
+        return string;
     }
 
     @Override
     public void setString(String string) {
+
+        this.string = string;
     }
 
     @Override
     public int count() {
-        return 0;
+
+        if (string == null){
+            return 0;
+        }
+
+        String[] words = string.trim().split("\\s+");
+        return words.length;
     }
 
     @Override
     public String removeNthCharacter(int n, boolean maintainSpacing) {
-        return null;
+        if (n <= 0) {
+            throw new IllegalArgumentException("Invalid value for n: " + n);
+        }
+
+        StringBuilder result = new StringBuilder();
+        int count = 0;
+
+        for (int i = 0; i < string.length(); i++) {
+            char c = string.charAt(i);
+            if ((i + 1) % n != 0) {
+                result.append(c);
+            } else if (maintainSpacing) {
+                result.append(" ");
+                count++;
+            }
+        }
+
+        if (!maintainSpacing && (string.length() % n) == 0) {
+            result.deleteCharAt(result.length() - 1);
+        }
+
+        return result.toString();
     }
 
+
+
     @Override
     public String[] getSubStrings(int startWord, int endWord) {
-        return null;
+        String[] words = string.split("\\s+");
+        int numWords = words.length;
+
+        if (startWord <= 0 || endWord <= 0 || startWord > endWord) {
+            throw new IllegalArgumentException("Invalid start or end word");
```

Comment:
I ran the test coverage on my local machine to see what test coverage issues jacoco is complaining about.
I was not able to upload a screen shot, but basically it identifies those lines that throw an exception as not being accounted for. Check you test cases to see if you check for those exceptions.